servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -225,12 +235,18 @@ StreamingHttpClient build() {
 
         ServiceDiscoverer<U, R, ? extends ServiceDiscovererEvent<R>> serviceDiscoverer(
                 HttpExecutionContext executionContext) {
+            BiIntFunction<Throwable, ? extends Completable> sdRetryStrategy = serviceDiscovererRetryStrategy;
+            if (sdRetryStrategy == null && deprecatedServiceDiscovererRetryStrategy == null) {
+                sdRetryStrategy = retryWithConstantBackoffDeltaJitter(__ -> true, SD_RETRY_STRATEGY_INIT_DURATION,
+                        SD_RETRY_STRATEGY_JITTER, executionContext.executor());
+            }
             return new RetryingServiceDiscoverer<>(new StatusAwareServiceDiscoverer<>(sd, sdStatus),
-                            serviceDiscovererRetryStrategy == null ?
-                                    DefaultServiceDiscoveryRetryStrategy.Builder
-                                            .<R>withDefaults(executionContext.executor(),
-                                                    SD_RETRY_STRATEGY_INIT_DURATION, SD_RETRY_STRATEGY_JITTER).build() :
-                                    serviceDiscovererRetryStrategy);
+                    sdRetryStrategy,
+                    deprecatedServiceDiscovererRetryStrategy == null ?
+                            DefaultServiceDiscoveryRetryStrategy.Builder
+                                    .<R>withDefaults(executionContext.executor(),
```

Comment:
We should avoid building `DefaultServiceDiscoveryRetryStrategy`, it should be used only if a custom one is provided by users.